### ISSUE TITLE
Should checkboxes support true/false?

### ISSUE THREAD

**Issue by nkylstad** [Altinn/app-frontend-react#114]

**Comment by Hannott** [comment]
In my mind this is asking if a thing should do what it was made to do. While it technically has three states, in practice it's either true or false, and null should be treated as false, or you should be using a different component.

The component as it is now should be renamed "multiple choice checkboxes", and the standard checkbox component should work like a standard checkbox, true/false

**Comment by FinnurO** [comment]
It seems clear that, according to the HTML specification https://html.spec.whatwg.org/#checkbox-state-(type=checkbox), we should only support true/false and not a third state. 

So the tasks should be to follow the standard and implement boolean support as commented.

**Comment by mjulstein** [comment]
@FinnurO It is true that the spec only has two states, but the [indeterminate](https://html.spec.whatwg.org/#dom-input-indeterminate) IDL attribute, should possibly used when the user has to make a decision and has not yet done so?

**Comment by mjulstein** [comment]
I think 2 states for the checkbox. A toggle is a radiobitton.
There is an initial undefined state that can be in store though, if the user is either answering false or has not answered (undefined) the check can be unchecked. If the answer is true it is checked.
The html component has only 2 states, but the form in redux can have as many as we want.

**Comment by SimenRokaas** [comment]
I currently got a change request to replace radio buttons for a yes/no boolean field with a checkbox, and to my big surprise I find that it is not possible?! 

I totally agree with @Hannott above. Not supporting a simple yes/no boolean field as a checkbox seems very strange! Please do implement this 🙏

**Comment by olemartinorg** [comment]
I also agree, that's not the problem. Currently, app-frontend-react does not support any other data types than strings in the data model, so before we can have booleans anywhere we need to solve #205. As that's a breaking change, we would need to release a new major version (because manually testing your app is needed, and possibly adjusting custom code/backend in the process). Work has started, but it's currently not progressing quite as much as we'd like, as there are other tasks we need to complete in order to make that change manageable to implement (such as the one I'm working on right now, #377). In the end, when we get there, the plan is to implement the use-case for a single boolean-bound checkbox in #681.

**Comment by altinnadmin** [comment]
@olemartinorg  Is there a reason why we can't get non-breaking #681 out of the door quickly, to solve most simple cases out there and take away "heat" from this issue?

**Comment by olemartinorg** [comment]
I guess we could _try_, but it depends on how the backend/platform handles our broken serialization. Right now, everything we send from app-frontend to the backend (app-lib-dotnet) is strings inside JSON. When the backend converts that to XML in some way, I guess the JSON types don't really matter - but as long as the strongly typed C# backend types a boolean as an actual boolean (not a stringy `'true' | 'false'`), I suspect more and more things will subtly break on the backend if we put the horse before the cart on this one.

Right now, the progress on our app-frontend `v4` (that, among other things, aims to introduce support for multiple types) is moving slower than I'd like. It's the same story as always; more urgent matters get prioritized, and it's difficult to see the long-term benefits of such structural changes.
